lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +855,126 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+// \returns true if all dimensions of the input tensors are the same except
+// for the provided dimension, otherwise return false.
```

Comment:
The method does not verify the node. It checks if all dimensions are the same except \p dim. A better name would be isConcatNodeUniformDims or something like that.

Reply:
done